In [ ]:
## how to load in CSVs

# Study data files
mouse_metadata_path = "data/Mouse_metadata.csv"
study_results_path = "data/Study_results.csv"

# Read the mouse data and the study results
mouse_metadata = pd.read_csv(mouse_metadata_path)
study_results = pd.read_csv(study_results_path)



In [106]:
import requests
import json
import pandas as pd
import numpy as np
from numpy import random 
import matplotlib.pyplot as plt
from scipy.stats import levene
import scipy.stats as stats

from collections import deque
from scipy.stats import linregress
from scipy.stats import wilcoxon
from scipy import stats

# Network Analyses


In [8]:
# Study data files

AllNetworks = "Data/NetworkRatingData.csv"

# Read the Network data and the study results

allnetworkdata = pd.read_csv(AllNetworks)

In [10]:
print(len(allnetworkdata))

11804


In [21]:

allnetworkdata = allnetworkdata.dropna(how="any")
allnetworkdata.head(30)

IDs            Networks  Ratings
0     1                 CBS      6.5
1     2                 CBS      8.9
2     3  CTV Sci-Fi Channel      7.5
3     4              The CW      7.4
4     5                 HBO      8.3
5     6              The CW      7.7
6     7            Showtime      8.2
7     8                 FOX      6.8
8     9                 ABC      7.8
9    10                 NBC      8.5
10   11                 FOX      7.7
11   12            Showcase      8.1
12   13              The CW      7.8
13   14            Showcase      8.2
14   15                 NBC      7.5
15   16            Showtime      8.3
16   18                 CBS      7.8
17   19              The CW      8.4
18   20                  FX      7.5
19   21                 TNT      7.5
20   22                 HBO      7.8
21   23                 ABC      6.7
22   24                 CBS      7.6
23   25             Fuji TV      7.8
24   26                 TBS      8.3
25   27                 NTV      8.8
26   28            Showtime      8.0
27   29             History      8.7
28   30                  FX      7.8
29   31                 ABC      8.0

In [22]:
#Summary Statistics 

mean=allnetworkdata.groupby(["Networks"]).mean()["Ratings"]

analysis_df=pd.DataFrame({"mean network rating":mean})

# Format

analysis_df=analysis_df.round(2)

analysis_df

mean network rating
Networks                       
1+1                        7.00
10 Peach                   7.50
A&E                        7.31
ABC                        7.32
ABC Comedy                 7.50
...                         ...
Россия 1                   7.53
СТС                        5.57
ТВ-3                       5.90
ТНТ                        7.18
ЦТ СССР                    8.15

[219 rows x 1 columns]

In [23]:
analysis_df.sort_values(by=['mean network rating'], inplace=True, ascending=False)
analysis_df.head()

#clean
clean_analysis_df = analysis_df.dropna(how="any")
clean_analysis_df.head()

mean network rating
Networks                              
Star TV                            9.8
Smithsonian                        9.5
Channel 9 MCOT HD                  9.2
France 2                           9.2
ESPN2                              9.0

In [24]:
#Top 20 rated networks

first20_networks =clean_analysis_df.head(20)

first20_networks = pd.DataFrame(first20_networks)
newfirst20_networks = first20_networks.reset_index().set_index('Networks')
newfirst20_networks

mean network rating
Networks                               
Star TV                            9.80
Smithsonian                        9.50
Channel 9 MCOT HD                  9.20
France 2                           9.20
ESPN2                              9.00
Lifetime Movies                    8.80
CBS Sports Network                 8.80
BBC Scotland                       8.80
SUN-TV                             8.80
ZDF                                8.70
NDR                                8.50
TVNorge                            8.50
H2                                 8.50
Fox Kids                           8.45
Das Erste                          8.40
Channel 10                         8.40
TVNZ 1                             8.30
Create                             8.30
History Asia                       8.30
Really                             8.30

In [25]:
#Lowest rated networks

last20_networks =clean_analysis_df.tail(20)

last20_networks = pd.DataFrame(last20_networks)
newlast20_networks = last20_networks.reset_index().set_index('Networks')
newlast20_networks

mean network rating
Networks                             
Quest                            6.30
TLC                              6.27
VTM                              6.20
Pop                              6.15
E!                               6.04
France 4                         6.00
AXS TV                           6.00
City                             6.00
ТВ-3                             5.90
Kanal 5                          5.70
RTL                              5.70
СТС                              5.57
Gold                             5.55
UP TV                            5.45
MSNBC                            5.35
nick@nite                        5.20
FOX News Channel                 5.07
TVP2                             4.80
CNN                              4.74
ITV Be                           3.00

In [109]:
ratingslast20=newlast20_networks['mean network rating']
ratingsfirst20=newfirst20_networks['mean network rating']

In [67]:

#pulling the top mean network names to do an ANOVA on these networks
NetworkslistLast20 = newlast20_networks.index.tolist()
Networkslistfirst20 = newfirst20_networks.index.tolist()

#### Last 20 Networks Analysis

In [82]:
#ANOVA for Last 20

Network11 = allnetworkdata[allnetworkdata['Networks'] == NetworkslistLast20[0]]['Ratings']
Network21 = allnetworkdata[allnetworkdata['Networks'] == NetworkslistLast20[1]]['Ratings']
Network31 = allnetworkdata[allnetworkdata['Networks'] == NetworkslistLast20[2]]['Ratings']
Network41 = allnetworkdata[allnetworkdata['Networks'] == NetworkslistLast20[3]]['Ratings']
Network51 = allnetworkdata[allnetworkdata['Networks'] == NetworkslistLast20[4]]['Ratings']
Network61 = allnetworkdata[allnetworkdata['Networks'] == NetworkslistLast20[5]]['Ratings']
Network71 = allnetworkdata[allnetworkdata['Networks'] == NetworkslistLast20[6]]['Ratings']
Network81 = allnetworkdata[allnetworkdata['Networks'] == NetworkslistLast20[7]]['Ratings']
Network91 = allnetworkdata[allnetworkdata['Networks'] == NetworkslistLast20[8]]['Ratings']
Network101 = allnetworkdata[allnetworkdata['Networks'] == NetworkslistLast20[9]]['Ratings']
Network111 = allnetworkdata[allnetworkdata['Networks'] == NetworkslistLast20[10]]['Ratings']
Network121 = allnetworkdata[allnetworkdata['Networks'] == NetworkslistLast20[11]]['Ratings']
Network131 = allnetworkdata[allnetworkdata['Networks'] == NetworkslistLast20[12]]['Ratings']
Network141 = allnetworkdata[allnetworkdata['Networks'] == NetworkslistLast20[13]]['Ratings']
Network151 = allnetworkdata[allnetworkdata['Networks'] == NetworkslistLast20[14]]['Ratings']
Network161 = allnetworkdata[allnetworkdata['Networks'] == NetworkslistLast20[15]]['Ratings']
Network171 = allnetworkdata[allnetworkdata['Networks'] == NetworkslistLast20[16]]['Ratings']
Network181 = allnetworkdata[allnetworkdata['Networks'] == NetworkslistLast20[17]]['Ratings']
Network191 = allnetworkdata[allnetworkdata['Networks'] == NetworkslistLast20[18]]['Ratings']
Network201 = allnetworkdata[allnetworkdata['Networks'] == NetworkslistLast20[19]]['Ratings']


In [83]:
stat, p = levene(Network11, Network21, Network31,Network41, Network51, Network61,Network71, Network81, Network91,Network101, Network111, Network121,Network131, Network141, Network151,Network161, Network171, Network181,Network191, Network201)
print(f"The p value is: {p}. We fail to reject the null hypothesis that the data do not differ significantly from normality.")

The p value is: 0.6244774130677446. We fail to reject the null hypothesis that the data do not differ significantly from normality.


In [84]:
stats.f_oneway(Network11, Network21, Network31,Network41, Network51, Network61,Network71, Network81, Network91,Network101, Network111, Network121,Network131, Network141, Network151,Network161, Network171, Network181,Network191, Network201)

F_onewayResult(statistic=0.28542618418362137, pvalue=0.9969904045719367)

The ratings of the 20 lowest rated networks were compared using a one-way ANOVA. No significant differences were found. (F(?DF) = 0.285, p=0.99).

#### First 20 Networks Analysis

In [85]:
Network = allnetworkdata[allnetworkdata['Networks'] == Networkslistfirst20[0]]['Ratings']
Network2 = allnetworkdata[allnetworkdata['Networks'] == Networkslistfirst20[1]]['Ratings']
Network3 = allnetworkdata[allnetworkdata['Networks'] == Networkslistfirst20[2]]['Ratings']
Network4 = allnetworkdata[allnetworkdata['Networks'] == Networkslistfirst20[3]]['Ratings']
Network5 = allnetworkdata[allnetworkdata['Networks'] == Networkslistfirst20[4]]['Ratings']
Network6 = allnetworkdata[allnetworkdata['Networks'] == Networkslistfirst20[5]]['Ratings']
Network7 = allnetworkdata[allnetworkdata['Networks'] == Networkslistfirst20[6]]['Ratings']
Network8 = allnetworkdata[allnetworkdata['Networks'] == Networkslistfirst20[7]]['Ratings']
Network9 = allnetworkdata[allnetworkdata['Networks'] == Networkslistfirst20[8]]['Ratings']
Network10 = allnetworkdata[allnetworkdata['Networks'] == Networkslistfirst20[9]]['Ratings']
Network11 = allnetworkdata[allnetworkdata['Networks'] == Networkslistfirst20[10]]['Ratings']
Network12 = allnetworkdata[allnetworkdata['Networks'] == Networkslistfirst20[11]]['Ratings']
Network13 = allnetworkdata[allnetworkdata['Networks'] == Networkslistfirst20[12]]['Ratings']
Network14 = allnetworkdata[allnetworkdata['Networks'] == Networkslistfirst20[13]]['Ratings']
Network15 = allnetworkdata[allnetworkdata['Networks'] == Networkslistfirst20[14]]['Ratings']
Network16 = allnetworkdata[allnetworkdata['Networks'] == Networkslistfirst20[15]]['Ratings']
Network17 = allnetworkdata[allnetworkdata['Networks'] == Networkslistfirst20[16]]['Ratings']
Network18 = allnetworkdata[allnetworkdata['Networks'] == Networkslistfirst20[17]]['Ratings']
Network19 = allnetworkdata[allnetworkdata['Networks'] == Networkslistfirst20[18]]['Ratings']
Network20 = allnetworkdata[allnetworkdata['Networks'] == Networkslistfirst20[19]]['Ratings']

In [86]:
stat, p = levene(Network1, Network2, Network3,Network4, Network5, Network6,Network7, Network8, Network9,Network10, Network11, Network12,Network13, Network14, Network15,Network16, Network17, Network18,Network19, Network20)
print(f"The p value is: {p}. We reject the null hypothesis that the data do not differ significantly from normality.")

The p value is: 7.766198643075823e-44. We reject the null hypothesis that the data do not differ significantly from normality.


In [87]:
stats.kruskal(Network1, Network2, Network3,Network4, Network5, Network6,Network7, Network8, Network9,Network10, Network11, Network12,Network13, Network14, Network15,Network16, Network17, Network18,Network19, Network20)

KruskalResult(statistic=15.378289473684205, pvalue=0.698270108857485)

The ratings of the 20 highest rated networks were compared using a Kruskal-Wallis H Test. No significant differences were found. (H(?DF) = 15.37, p=0.69).

#### Differences between the average rating of the top 20 and last 20 networks analysis

In [112]:
stat, p = levene(ratingsfirst20, ratingslast20)
print(f"The p value is: {p}. We fail to reject the null hypothesis that the data do not differ significantly from normality.")

The p value is: 0.23516341365909896. We fail to reject the null hypothesis that the data do not differ significantly from normality.


In [110]:
stats.ttest_ind(ratingsfirst20, ratingslast20, equal_var=False)

Ttest_indResult(statistic=16.285735379854053, pvalue=2.0840968904129293e-16)

In [115]:
print(ratingsfirst20.mean())
print(ratingslast20.mean())
print(ratingsfirst20.std())
print(ratingslast20.std())

8.727500000000003
5.5495
0.4271924871210066
0.7609860293343796


The ratings of the top and bottom 20 rated networks were compared with an independent samples t-test. The top 20 networks (M=8.7275, SD=0.427) and the last 20 networks (M=5.549, SD=0.760) were significantly different (t(?DF)= 16.28, p=2.084e-16). 

# TV Show Analyses

In [15]:
# Study data files

first20show = "Data/First20Shows.csv"

# Read the show data and the study results

top20shows = pd.read_csv(first20show)

In [16]:
top20shows.head()

Ratings
0      9.8
1      9.8
2      9.5
3      9.5
4      9.4